internal/renderer/renderer.go

```diff
@@ -0,0 +1,63 @@
+package renderer
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/docker/app/internal/renderer/driver"
+	"github.com/pkg/errors"
+)
+
+var (
+	driversMu sync.RWMutex
+	drivers   = map[string]driver.Driver{}
+)
+
+// Register makes a renderer available by the provided name.
+// If Register is called twice with the same name or if driver is nil,
+// it panics.
+func Register(name string, driver driver.Driver) {
+	driversMu.Lock()
+	defer driversMu.Unlock()
+	if driver == nil {
+		panic("renderer: Register driver is nil")
+	}
+	if _, dup := drivers[name]; dup {
+		panic("renderer: Register called twice for driver " + name)
+	}
+	drivers[name] = driver
+}
+
+// Apply applies the specified render to the specified string with the specified settings.
+// If the render is not present is the registered ones, it errors out.
+func Apply(s string, settings map[string]interface{}, renderers ...string) (string, error) {
+	var err error
+	for _, r := range renderers {
+		if r == "none" {
+			continue
+		}
+		driversMu.RLock()
+		d, present := drivers[r]
+		driversMu.RUnlock()
+		if !present {
+			return "", errors.Errorf("unknown renderer %s", r)
+		}
+		s, err = d.Apply(s, settings)
+		if err != nil {
+			return "", err
+		}
+	}
+	return s, nil
+}
+
+// Drivers returns a sorted list of the names of the registered drivers.
+func Drivers() []string {
+	driversMu.RLock()
+	defer driversMu.RUnlock()
+	list := []string{"none"}
+	for name := range drivers {
+		list = append(list, name)
+	}
+	sort.Strings(list)
```

Comment:
nit: the sorting could be made after releasing the lock, but given that we only have a handful of values it doesn't really matter in practice.

Reply:
RLock and RUnlock should be just around the for loop.